### PR TITLE
Major changes to plotting scripts from eva source code

### PR DESCRIPTION
**Pull request by kevindougherty-noaa**
These changes were made when developing source code for [eva](https://github.com/danholdaway/eva). The goal of these changes was to separate the code into 3 levels:

image

This structure solved issues we had in the past including creating subplots. Plot types remain objects, however plots themselves are now objects that then can be created on a single figure. Shown here is some Psuedo Code on how it would work:

image

Will close outstanding issue #94

### CODE REVIEW

**Comment by kevindougherty-noaa** [comment]
@CoryMartin-NOAA I expect the tests to fail because I have not updated the example plot scripts. However, it was failing in the `setup.cfg` saying Cartopy needed Proj version 8.0.0. I tried to add that to the install requires in the `setup.cfg` and getting another error kicked back. Any thoughts?

**Comment by CoryMartin-NOAA** [comment]
@kevindougherty-noaa ugh yes, @danholdaway and I struggled with this for a while. We need to update the CI in EMCPy to match what EVA has now.

**Comment by danholdaway** [comment]
I made a workaround in this workflow file https://github.com/danholdaway/eva/blob/develop/.github/workflows/eva_tests_application.yml Note that it relies on a special `requirements-github.txt` file for the installation.

**Comment by kevindougherty-noaa** [comment]
Github actions are now getting to tests, hooray! Let's get the tests running with the new updated code.

**Comment by kevindougherty-noaa** [comment]
Finally figured out how to get the pytests to run and pass. Please ignore the several commits that do nothing :)

If you all can review the code changes, that would be great. Also a note, some things have been left out in the changes (dual x/y axes, plot logo, etc.) These will be added back in to fit the new code structure in separate PRs.

**Comment by danholdaway** [comment]
Out of curiosity...is the intention here that we would be able to remove EMCpy from Eva after this is merged?

**Comment by CoryMartin-NOAA** [comment]
@danholdaway eventually, but not yet. @kevindougherty-noaa still wants to do a few more things to get them in sync, and then we will remove the plotting from eva, add a git submodule, and adjust `import` as needed. Is that okay with you @danholdaway ?

**Comment by CoryMartin-NOAA** [comment]
also @danholdaway I would like to make you a maintainer of this repo if you want, but I need to ask @aerorahul to add you.

File: src/emcpy/plots/create_plots.py
Comment (EdwardSafford-NOAA) [inline src/emcpy/plots/create_plots.py]: Very minor -- the comment isn't entirely clear.  Maybe "checks to see if linear regression attribute exists" or something similar.

File: src/emcpy/plots/create_plots.py
Comment (EdwardSafford-NOAA) [inline src/emcpy/plots/create_plots.py]: That should probably be string instead of sting.

**Review by EdwardSafford-NOAA — DISMISSED** [review]
Looks good to me.  I noted a couple of small typos.

**Comment by kevindougherty-noaa** [comment]
I am not sure as to why the tests are now failing after my latest commit when they were passing prior. All that was changed were two lines of comments and no actual code. Does anyone have any idea as to what may be happening?

**Comment by danholdaway** [comment]
@kevindougherty-noaa I've seen a similar thing starting to fail in Eva too. I wonder if it's because some version has changes under us on the GitHub server side?

**Comment by kevindougherty-noaa** [comment]
@aerorahul Would you have any insight as to what may be causing these Github actions tests to fail? It seems to be related only to plots using `cartopy`.

**Comment by kevindougherty-noaa** [comment]
@danholdaway Your fix in eva worked for here as well. Thank you for figuring that out!

**Review by EdwardSafford-NOAA — APPROVED** [review]
Looks good.

**Pull request merged**
